backend/lambda/function.mjs

```diff
@@ -39,66 +49,114 @@ export const handler = async (event, context) => {
     try {
         switch (event.requestContext.routeKey) {
             case "POST /levels":
-                // TODO Actual Implementation
                 requestJSON = JSON.parse(event.body);
-                var levelName = requestJSON.name; 
-                body = {
+                var levelName = requestJSON.name;
+                if (!levelName) throw new BadRequestException(`'name' must be provided in the request.`);
+                var levelCreator = requestJSON.creator;
+                if (!levelCreator) throw new BadRequestException(`'creator' must be provided in the request.`);
+                var levelCreatorId = levelCreator.id;
+                if (!levelCreatorId) throw new BadRequestException(`'creator.id' must be provided in the request.`);
+                var levelCreatorName = levelCreator.name;
+                if (!levelCreatorName) throw new BadRequestException(`'creator.name' must be provided in the request.`);
+                // TODO Further validation of creator - maybe checking that the ID even exists?
+                var levelStatus = requestJSON.status;
+                if (!levelStatus) throw new BadRequestException(`'status' must be provided in the request.`);
+                // TODO Further validation of status (ensuring it is a valid one)
+                var levelData = requestJSON.data; 
+                if (!levelData) throw new BadRequestException(`'data' must be provided in the request.`);
+                
+                var generatedLevelId = uuidv4();
+                var currentTimestamp = Date.now();
+
```

Comment:
What do we do with `generatedLevelId`? Do we need to return that to the game so it can be stored for future requests?

Reply:
That'll be for specific GET requests - getting individual levels. The list query is for populating the list views (see all the latest published levels, see all the levels for a certain creator) (https://github.com/LPGameDevs/EditarrrPublic/blob/develop/backend/README.md#get-levels)

_Then_, when someone selects a level, you click on that level, take that ID, and provide it in a singular GET request: https://github.com/LPGameDevs/EditarrrPublic/blob/develop/backend/README.md#get-level (this response has all the level data). Makes it easier on the DB and the response size to not include all the level data in the plural request

Reply:
We can probably be selective with the get levels request, ie not show all the level data, just stuff like code/creator/etc